src/blueprints/tv2/factories/tv2-transition-factory.ts

```diff
@@ -0,0 +1,110 @@
+import { AtemMeTimelineObject, AtemTransition, AtemType } from '../../timeline-state-resolver-types/atem-types'
+import { DeviceType } from '../../../model/enums/device-type'
+import { Tv2BlueprintTimelineObject } from '../value-objects/tv2-meta-data'
+import { TimelineObject } from '../../../model/entities/timeline-object'
+
+const FRAME_RATE: number = 25
+
+const FRAMES_FOR_MIX_TRANSITION: number = 25 // TODO: Get from configuration.
+
+const MIX_TRANSITION_ID: string = 'nextTakeHasMixTransitionAction'
+
+export class Tv2TransitionFactory {
+
+  public isTransitionAction(action: Action): boolean {
+    return [MIX_TRANSITION_ID].includes(action.id)
+  }
+
+  public getMutateActionMethods(action: Action): MutateActionMethods | undefined {
+    switch (action.id) {
+      case MIX_TRANSITION_ID: {
+        return {
+          updateActionWithPlannedPieceData: (action: Action, plannedPiece: Piece) => this.updateAtemMeInput(action, plannedPiece),
+          plannedPiecePredicate: (piece: Piece) => piece.timelineObjects.some(timelineObject => timelineObject.layer === Tv2AtemLayer.PROGRAM)
+        }
+      }
+    }
+  }
+
+  public createMixTransitionAction(): PieceAction {
+    const pieceInterface: PieceInterface = {
+      id: 'mixTransitionActionPiece',
+      name: 'Mix transition',
+      partId: '',
+      type: PieceType.TRANSITION,
+      layer: Tv2SourceLayer.JINGLE,
+      pieceLifespan: PieceLifespan.WITHIN_PART,
+      transitionType: TransitionType.IN_TRANSITION,
+      isPlanned: false,
+      start: 0,
+      duration: this.getTimeFromFrames(FRAMES_FOR_MIX_TRANSITION),
+      postRollDuration: 0,
+      preRollDuration: 0,
+      tags: [],
+      timelineObjects: []
+    }
+
+    return {
+      id: MIX_TRANSITION_ID,
+      name: 'Mix transition on next take',
+      type: ActionType.INSERT_PIECE_AS_NEXT,
+      data: pieceInterface
+    }
+  }
+
+  private getTimeFromFrames(frames: number): number {
+    return (1000 / FRAME_RATE) * frames
+  }
+
+  private createAtemMixTransitionTimelineObject(sourceInput: number): AtemMeTimelineObject {
+    return {
+      id: '',
+      enable: {
+        start: 0
+      },
+      layer: Tv2AtemLayer.PROGRAM,
+      priority: 10,
+      content: {
+        deviceType: DeviceType.ATEM,
+        type: AtemType.ME,
+        me: {
+          input: sourceInput,
+          transition: AtemTransition.MIX,
+          transitionSettings: {
+            mix: {
+              rate: FRAMES_FOR_MIX_TRANSITION
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private updateAtemMeInput(action: Action, plannedPiece: Piece): Action {
+    const timelineObject: TimelineObject | undefined = plannedPiece.timelineObjects.find(timelineObject => timelineObject.layer === Tv2AtemLayer.PROGRAM)
+    if (!timelineObject) {
+      console.log(`Can't update Atem Me Input. No TimelineObject for '${Tv2AtemLayer.PROGRAM}' found on Piece '${plannedPiece.id}'.`)
```

Comment:
Small thing. When we update the log statements to use Tv2's logging NPM package, should this be an info log or a warning log?

If it should be a warning log, maybe consider using `console.warn(...)` for it. That way it easy to see waht logging type is intended to be used, for when the update to the NPM package.

Same goes for the log statement in the next if block, and perhabs any other log statements elsewhere.

Reply:
Honestly, the only I thing I thought about here was that I thought it was a bit too agressive to throw an error, hence the log. 

In general, I'm more in favour of throwing errors then just logging statement when something goes wrong. 
Personally, I think we should determine if this scenario shouldn't be possible, and hence throw an error. If the scenario is possible, then it's a valid use case and we shouldn't log at all.

Reply:
I also favour throwing errors, but as you said, that is sometimes too aggresive.
When throwing an error is too aggresive, logging a warning, would be my go-to way of indicating that something isn't quite right, but not critically so.

If in the end it is decided to not throw an error, but it still isn't a valid use case, then i'd very much like for it to be updated to a `console.warn(...)`.